apps/client/app/routes/_layout.emails.tsx

```diff
@@ -0,0 +1,114 @@
+import { Form, useLoaderData, useNavigate, useNavigation, useSubmit } from "@remix-run/react";
+import { json, LoaderFunctionArgs } from "@remix-run/node";
+import { useEffect } from "react";
+import { MOCKED_EMAILS } from "../../mocks/emails";
+import { Email } from "../../models/email.model";
+
+
+async function getEmails(searchString: string | null): Promise<Email[]> {
+  await new Promise((resolve) => setTimeout(resolve, 500));
+  if (searchString?.length) {
+    return MOCKED_EMAILS.filter((item) => item.title.toLowerCase().includes(searchString.toLowerCase()));
+  }
+  return MOCKED_EMAILS;
+}
+
+export const loader = async ({
+                               request,
+                             }: LoaderFunctionArgs) => {
+  const url = new URL(request.url);
+  const searchedStr = url.searchParams.get("email");
```

Comment:
Remix has a built in `useSearchParams` hook for this purpose

Reply:
Just took a look at the docs and I think in this case because we are accessing the request.url on the server (this is running inside the loader function) I don't think we can use the useSearchParams hook.

https://remix.run/docs/en/main/route/loader

Reply:
Ah good catch, I will mark this as resolved